packages/contracts-bedrock/src/L1/ProtocolVersions.sol

```diff
@@ -0,0 +1,131 @@
+
+/// @title ProtocolVersions
+/// @notice The ProtocolVersions contract is used to manage superchain protocol version information.
+contract ProtocolVersions is OwnableUpgradeable, ISemver {
+    /// @notice Enum representing different types of updates.
+    /// @custom:value REQUIRED_PROTOCOL_VERSION              Represents an update to the required protocol version.
+    /// @custom:value RECOMMENDED_PROTOCOL_VERSION           Represents an update to the recommended protocol version.
+    enum UpdateType {
+        REQUIRED_PROTOCOL_VERSION,
+        RECOMMENDED_PROTOCOL_VERSION
+    }
+
+    /// @notice Version identifier, used for upgrades.
+    uint256 public constant VERSION = 0;
+
+    /// @notice Storage slot that the required protocol version is stored at.
+    bytes32 public constant REQUIRED_SLOT = bytes32(uint256(keccak256("protocolversion.required")) - 1);
+
+    /// @notice Storage slot that the recommended protocol version is stored at.
+    bytes32 public constant RECOMMENDED_SLOT = bytes32(uint256(keccak256("protocolversion.recommended")) - 1);
+
+    /// @notice Emitted when configuration is updated.
+    /// @param version    ProtocolVersion version.
+    /// @param updateType Type of update.
+    /// @param data       Encoded update data.
+    event ConfigUpdate(uint256 indexed version, UpdateType indexed updateType, bytes data);
+
+    /// @notice Semantic version.
+    /// @custom:semver 0.1.0
+    string public constant version = "0.1.0";
+
+    /// @notice Constructs the ProtocolVersion contract. Cannot set
+    ///         the owner to `address(0)` due to the Ownable contract's
+    ///         implementation, so set it to `address(0xdEaD)`
+    ///         A zero version is considered empty and is ignored by nodes.
+    constructor() {
+        initialize({
+            _owner: address(0xdEaD),
+            _required: ProtocolVersion.wrap(uint256(0)),
+            _recommended: ProtocolVersion.wrap(uint256(0))
+        });
+    }
+
+    /// @notice Initializer.
+    /// @param _owner             Initial owner of the contract.
+    /// @param _required          Required protocol version to operate on this chain.
+    /// @param _recommended       Recommended protocol version to operate on thi chain.
+    function initialize(
+        address _owner,
+        ProtocolVersion _required,
+        ProtocolVersion _recommended
+    )
+        public
+        reinitializer(2)
+    {
+        __Ownable_init();
+        transferOwnership(_owner);
+        _setRequired(_required);
+        _setRecommended(_recommended);
+    }
+
+    /// @notice Returns a ProtocolVersion stored in an arbitrary storage slot.
+    ///         These storage slots decouple the storage layout from solc's automation.
+    /// @param _slot The storage slot to retrieve the address from.
+    function _getProtocolVersion(bytes32 _slot) internal view returns (ProtocolVersion out_) {
+        assembly {
+            out_ := sload(_slot)
```

Comment:
same as above, why not just inline this into the other two places its used?

Reply:
See other comment